lib/rules/no-console.js

```diff
@@ -102,10 +105,31 @@ module.exports = {
         function report(reference) {
             const node = reference.identifier.parent;
 
+            const propertyName = astUtils.getStaticPropertyName(node);
+
             context.report({
                 node,
                 loc: node.loc,
-                messageId: "unexpected"
+                messageId: "unexpected",
+                suggest: [
+                    {
+                        messageId: "removeConsole",
+                        data: { propertyName },
+                        fix(fixer) {
+                            const copyOfStatementListParents = astUtils.STATEMENT_LIST_PARENTS;
+
+                            copyOfStatementListParents.delete("Program");
+
+                            const isNotFixable = copyOfStatementListParents.has(node.parent.parent.parent.type);
```

Comment:
It should be the opposite - we want to remove only statements that are in statement lists.

```js
if (foo) {
    console.log(foo); // yes
}

if (foo) console.log(foo); // no
```

We should also check if `node.parent` is CallExpression, `node` its callee, and if `node.parent.parent` is ExpressionStatement.

Reply:
Fixed in https://github.com/eslint/eslint/pull/17680/commits/00d5428dfa923e3917c405d5f1f33647913b6ee0.